rsopt/codes/radia/sim_functions_km.py

```diff
@@ -0,0 +1,143 @@
+import radia as rad
+import numpy as np
+from scipy.constants import mu_0, e, m_e, c
+
+
+def hybrid_undulator(lpx, lpy, lpz, pole_properties, pole_segmentation, pole_color,
```

Comment:
The `hybrid_undulator` `materials` and `undulatorK_simple` have existing implementations in rsopt/codes/radia/sim_functions.py. It would be better to modify that file and add any other needed Radia-dependent calculations there.